src/stages/main/patchers/ConstructorPatcher.js

```diff
@@ -73,6 +144,32 @@ export default class ConstructorPatcher extends ObjectBodyMemberPatcher {
     return -1;
   }
 
+  getIndexOfFirstThisStatement() {
+    if (!this.expression.body) {
+      return -1;
+    }
+    let statements = this.expression.body.statements;
+    for (let i = 0; i < statements.length; i++) {
+      let usesThis = false;
+      traverse(statements[i].node, child => {
+        if (usesThis) {
+          // Already found it, skip this one.
+          return false;
+        } else if (child.type === 'This') {
+          // Found it.
+          usesThis = true;
+        } else if (child.type === 'Class') {
+          // Don't go into other classes.
```

Comment:
Good point. Actually, I'll skip all functions to be consistent with babel's static analysis (we don't know when the function will be called).